src/main/java/sparksoniq/jsoniq/item/StringItem.java

```diff
@@ -76,6 +80,25 @@ public boolean isTypeOf(ItemType type) {
         return type.getType().equals(ItemTypes.StringItem) || super.isTypeOf(type);
     }
 
+    @Override
+    public boolean isCastableAs(AtomicType type) {
+        if (type.getType() == AtomicTypes.StringItem)
+            return true;
+        try {
+            if (type.getType() == AtomicTypes.IntegerItem)
+                Integer.parseInt(this._value);
+            else if (type.getType() == AtomicTypes.DecimalItem) {
+                if (this._value.contains("e") || this._value.contains("E")) return false;
+                Float.parseFloat(this._value);
+            } else if (type.getType() == AtomicTypes.DoubleItem)
+                Double.parseDouble(this._value);
+            else return isBooleanLiteral(this._value);
```

Comment:
And if type.getType() == AtomicTypes.NullItem, this works if the string is "null".